### DISCUSSION TITLE
Configure the frequency of updating the Tasks Status.

### DISCUSSION THREAD

**Discussion by roversnord** [Netflix/conductor#3229] [Q&A]
We have a simple task with timeoutSeconds and responseTimeoutSeconds configured and timeoutPolicy set to TIME_OUT_WF. When this simple task is picked up by the worker, it sets the status to in_progress and the callBackAfterSeconds(same value as the timeoutSeconds in the task definition). What I have noticed is the actual task timeout takes longer then the timeoutSeconds and it's always fluctuating. I have seen times ranging from 124 - 178 seconds. Wondering if there a parameter that controls how often and quickly the server updates the Task Status when there is a change in the status. The problem with the delay in updating the status is causing the worker to see the task in a non-terminal status and reprocess it.

Example of an instance when it timed out after 178 seconds when the timeout specified was 120.
`"reasonForIncompletion": "Task timed out after 178 seconds. Timeout configured as 120 seconds. Timeout policy configured to TIME_OUT_WF",`

**Comment by apanicker-nflx** [comment]
The timeout evaluation is currently a factor of the number of running workflows in the system. We are working on overhauling the running workflow evaluation framework to optimize timeout evaluations. This fix is expected to land sometime later this year.

In the meantime, you could increase the frequency of sweep and number of sweep threads as a stopgap solution.
